lib/python/qmk/path.py

```diff
@@ -33,3 +33,17 @@ def normpath(path):
         return os.path.normpath(path)
 
     return os.path.normpath(os.path.join(os.environ['ORIG_CWD'], path))
+
+
+def cpathfiles(dir_names):
+    """Returns a list of all *.c, *.h, and *.cpp files for a given list of directories
+
+    Args:
+
+        dir_names
+            List of directories, relative pathing starts at qmk's cwd
+    """
+    files = []
+    for dir in dir_names:
+        files.extend(Path(file) for file in Path(dir).glob('**/*') if file.suffix in ['.c', '.h', '.cpp'])
```

Comment:
Path.glob returns a generator for Path objects, so the `Path(file)` casting is not necessary.